src/ios/CDVCamera.m

```diff
@@ -402,6 +444,115 @@ - (NSString*)tempFilePath:(NSString*)extension
     return filePath;
 }
 
+// This is not absolute, but more mightNeedOrientationCorrection
+- (BOOL) needsOrientationCorrection:(UIImage*)image options:(CDVPictureOptions*)options
+{
+    // TODO use image to detect if it needs an orientation correction
+
+    // See FIXME #2
+    //    if (options.sourceType != UIImagePickerControllerSourceTypeCamera) {
+    //        return false;
+    //    }
+    return options.correctOrientation;
+}
+
+- (BOOL) needsResize:(CDVPictureOptions*)options
+{
+    return (options.targetSize.height > 0 && options.targetSize.width > 0);
+}
+
+- (BOOL) needsEdit:(UIImage*)image options:(CDVPictureOptions*)options
+{
+    return [self needsOrientationCorrection:image options:options] || [self needsResize:options];
+}
+
+- (BOOL) needsSavingToPhotoAlbum:(UIImage*)image options:(CDVPictureOptions*)options
+{
+    /*
+     We save to the photo album if:
+     - the option is set
+     - the image is fetch from the camera OR the image has been edited (no need to duplicate image in the library)
+     */
+    BOOL isSourceCamera = options.sourceType == UIImagePickerControllerSourceTypeCamera;
+    BOOL saveToPhotoAlbum = options.saveToPhotoAlbum && ([self needsEdit:image options:options] || isSourceCamera);
+    
+    return saveToPhotoAlbum;
+}
+
+/*
+ Metadata is not needed for:
+ - source: gallery
+ - destination: NATIVE_URI
+ - no edit (not orientation and not resize and not allowEdits)
+ 
+ otherwise, it can be found:
+ - source: camera
+ => UIImagePickerControllerMediaMetadata
+ - source: gallery
+ => CGImage thingy
```

Comment:
Sorry, forgot to remove those comments. They were only meant to remember how to access metadata in different situation (and might be totally outdated).